source/validate.cpp

```diff
@@ -520,24 +538,117 @@ spv_result_t ModuleLayoutPass(ValidationState_t& _,
               inst->words + inst->operands[2].offset;
           if (*storage_class != SpvStorageClassFunction)
             return _.diag(SPV_ERROR_INVALID_LAYOUT)
-                   << "All OpVariable instructions in a function must have a "
-                      "storage class of Function[7]";
+                   << "All Variable instructions in a function must have a "
+                      "storage class of function[7]";
+        } break;
+        default:
           break;
+      }
+      if (_.getLayoutStage() == kLayoutFunctionDeclarations) {
+        switch (opcode) {
+          case SpvOpFunction:
+            if (_.in_function_body()) {
```

Comment:
Ah, it looks like even a declaration has a function body, where parameters go.  Never mind...